openwisp_controller/geo/api/serializers.py

```diff
@@ -38,3 +33,120 @@ class Meta:
         model = Location
         geo_field = 'geometry'
         fields = '__all__'
+
+
+class BaseSerializer(FilterSerializerByOrgManaged, ValidatedModelSerializer):
+    pass
+
+
+class FloorPlanSerializer(BaseSerializer):
+    class Meta:
+        model = FloorPlan
+        fields = (
+            'id',
+            'floor',
+            'image',
+            'location',
+            'created',
+            'modified',
+        )
+        read_only_fields = ('created', 'modified')
+
+    def validate(self, data):
+        if data.get('location'):
+            data['organization'] = data.get('location').organization
+        instance = self.instance or self.Meta.model(**data)
+        instance.full_clean()
+        return data
+
+
+class LocationModelSerializer(BaseSerializer):
+    class Meta:
+        model = Location
+        fields = (
+            'id',
+            'organization',
+            'name',
+            'type',
+            'is_mobile',
+            'address',
+            'geometry',
+            'created',
+            'modified',
+        )
+        read_only_fields = ('created', 'modified')
+
+
+class NestedtLocationSerializer(gis_serializers.GeoFeatureModelSerializer):
+    class Meta:
+        model = Location
+        geo_field = 'geometry'
+        fields = (
+            'type',
+            'is_mobile',
+            'name',
+            'address',
+            'geometry',
+        )
+
+
+class NestedFloorplanSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = FloorPlan
+        fields = (
+            'floor',
+            'image',
+        )
+
+
+class DeviceLocationSerializer(serializers.ModelSerializer):
+    location = NestedtLocationSerializer()
+    floorplan = NestedFloorplanSerializer(required=False, allow_null=True)
```

Comment:
Hi, the representation of the data has been changed with this `gis_serializers.GeoFeatureModelSerializer` serializer which changes the hierarchy of the serializer fields, that result in the value not loading in the appropriate HTML form fields in the DRF's default Browsable Page, so the the page/form can be used for testing PUT requests, or to upload the image file, and then the further testing can be done with the JSON/form or OpenAPI Docs, page. I tried to find some generic way to render it, but since the data in the response, is sent in that manner, we'll need to either customize the DRF Base template, or have our own Custom page. OR we can have the representation in the default python primitive way.(which I don't think will be good)

PS: This is true for other packages which changes the representation of the object.